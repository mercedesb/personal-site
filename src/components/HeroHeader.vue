<template>
  <main>
    <div :class="'Hero Hero--' + color">
      <h2>{{ title }}</h2>
    </div>
  </main>
</template>

<script>
export default {
  props: {
    title: String,
    color: String
  }
}
</script>

<style lang="scss">
</style>