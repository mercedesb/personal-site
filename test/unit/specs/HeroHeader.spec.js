import HeroHeader from '@/components/HeroHeader.vue'
import { TestUtility } from './TestUtility'

describe('HeroHeader', () => {
  let component

  const initialProps = {
    title: 'hero title',
    color: 'brown',
    mainImage: 'http://google.com',
    backgroundImages: ['http://image1.com', 'http://image2.com', 'http://image3.com']
  }

  const shallow = propsData => TestUtility.shallow(HeroHeader, {
    propsData: {
      ...initialProps,
      ...propsData
    }
  })

  describe('Snapshots', () => {
    it('renders img for each item in props.backgroundImages', () => {
      component = shallow()
      expect(component.findAll('.Hero-background img')).toHaveLength(initialProps.backgroundImages.length)
    })

    describe('with a title', () => {
      it('matches snapshot', () => {
        component = shallow()
        expect(component.element).toMatchSnapshot()
      })
    })

    describe('without a title', () => {
      it('matches snapshot', () => {
        component = shallow({title: undefined})
        expect(component.element).toMatchSnapshot()
      })
    })
  })

  describe('Properties', () => {
    it('has a title property', () => {
      component = shallow()
      expect(component.props().title).toEqual('hero title')
    })

    it('has a color property', () => {
      component = shallow()
      expect(component.props().color).toEqual('brown')
    })

    it('has a mainImage property', () => {
      component = shallow()
      expect(component.props().mainImage).toEqual('http://google.com')
    })

    it('has a backgroundImages property', () => {
      component = shallow()
      expect(component.props().backgroundImages).toEqual(initialProps.backgroundImages)
    })
  })
})
