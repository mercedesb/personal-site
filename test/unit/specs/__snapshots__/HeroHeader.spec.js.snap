// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroHeader Snapshots with a title matches snapshot 1`] = `
<header
  class="Hero Hero--brown"
>
  <div
    class="Hero-contentContainer"
  >
    <div>
      <img
        class="Hero-image"
        src="http://google.com"
      />
       
      <!---->
    </div>
  </div>
   
  <div
    class="Hero-background"
  >
    <img
      class="Hero-backgroundImage"
      src="http://image1.com"
    />
    <img
      class="Hero-backgroundImage"
      src="http://image2.com"
    />
    <img
      class="Hero-backgroundImage"
      src="http://image3.com"
    />
  </div>
</header>
`;

exports[`HeroHeader Snapshots without a title matches snapshot 1`] = `<!---->`;
